src/room/useLoadGroupCall.ts

```diff
@@ -225,29 +225,52 @@ export const useLoadGroupCall = (
           });
         } else {
           // If the room does not exist we first search for it with viaServers
-          const roomSummary = await client.getRoomSummary(roomId, viaServers);
-          if (roomSummary.join_rule === JoinRule.Public) {
-            room = await client.joinRoom(roomSummary.room_id, {
+          let roomSummary: RoomSummary | undefined = undefined;
+          try {
+            roomSummary = await client.getRoomSummary(roomId, viaServers);
+          } catch (error) {
+            // If the room summary endpoint is not supported we let it be undefined and treat this case like
+            // `JoinRule.Public`.
+            // This is how the logic was done before: "we expect any room id passed to EC
+            // to be for a public call" Which is definitely not ideal but worth a try if fetching
+            // the summary crashes.
+            logger.error(
```

Comment:
```suggestion
            logger.warn(
```

A warning feels a little more appropriate since the application recovers gracefully and continues

Reply:
I wanted this to really standup so that there is a chance for this to be caught by users. In theory it should be considered as incompatible if a HS does not support the room summary endpoint.

But probably a warning is still enough.